engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheHashingJobActor.scala

```diff
@@ -0,0 +1,267 @@
+    Set(commandTemplateHash, backendNameHash, inputCountHash, outputCountHash) ++ runtimeAttributeHashes ++ inputHashResults ++ outputExpressionHashResults
+  }
+
+  private [callcaching] def makeFileHashingActor() = {
+    val fileHashingActorName = s"FileHashingActor_for_${jobDescriptor.key.tag}"
+    context.actorOf(fileHashingActorProps, fileHashingActorName)
+  }
+}
+
+object CallCacheHashingJobActor {
+
+  def props(jobDescriptor: BackendJobDescriptor,
+            callCacheReadingJobActor: Option[ActorRef],
+            initializationData: Option[BackendInitializationData],
+            runtimeAttributeDefinitions: Set[RuntimeAttributeDefinition],
+            backendName: String,
+            fileHashingActorProps: Props,
+            writeToCache: Boolean
+           ) = Props(new CallCacheHashingJobActor(
+    jobDescriptor,
+    callCacheReadingJobActor,
+    initializationData,
+    runtimeAttributeDefinitions,
+    backendName,
+    fileHashingActorProps,
+    writeToCache
+  ))
+
+  sealed trait CallCacheHashingJobActorState
+  case object WaitingForHashFileRequest extends CallCacheHashingJobActorState
+  case object HashingFiles extends CallCacheHashingJobActorState
+
+  /**
+    * Aggregates hash results together in a predictable and reproducible manner.
+    * Purposefully takes an Iterable, because it will be sorted appropriately in this method
+    * to ensure deterministic result, so the type of collection doesn't matter.
+    * Only aggregates hash values for which checkForHitOrMiss is true.
+    * If several hash keys are identical, the result of this method is undefined.
+    */
+  private def calculateHashAggregation(hashes: Iterable[HashResult], messageDigest: MessageDigest) = {
+    val sortedHashes = hashes.toList.filter(_.hashKey.checkForHitOrMiss).sortBy(_.hashKey.key).map(_.hashValue.value.getBytes)
+    sortedHashes foreach messageDigest.update
+    DatatypeConverter.printHexBinary(messageDigest.digest())
+  }
+
+  object CallCacheHashingJobActorData {
+    // Slick will eventually build a prepared statement with that many parameters. Don't set this too high or it will stackoverflow.
+    val BatchSize = 100
+
+    def apply(fileHashRequestsRemaining: List[SingleFileHashRequest], callCacheReadingJobActor: Option[ActorRef]) = {
+      new CallCacheHashingJobActorData(fileHashRequestsRemaining.grouped(BatchSize).toList, List.empty, callCacheReadingJobActor)
+    }
+  }
+
+  final case class CallCacheHashingJobActorData(
+                                           fileHashRequestsRemaining: List[List[SingleFileHashRequest]],
+                                           fileHashResults: List[HashResult],
+                                           callCacheReadingJobActor: Option[ActorRef]
+                                         ) {
+    private val md5Digest = MessageDigest.getInstance("MD5")
+
+    def withFileHash(hashResult: HashResult): (CallCacheHashingJobActorData, Option[CCHJAFileHashResponse]) = {
```

Comment:
It seems a bit superfluous for this one as this method is only ever used once inside this actor, and hence its result parsed once ? Also the name would be `NewDataAndMaybeResponse` :/
What about a scaladoc that says what the pair corresponds to ?